### PR TITLE
Allow socket throttle limits to be configurable

### PR DESCRIPTION
**Pull request by gordonwang0**
A better solution to https://github.com/Azure/iot-identity-service/issues/433 is to add a config option to allow the throttling limits to be configurable. This PR adds the `max_requests` option, which controls the throttling limit.

Reverts #438 and removes the no-socket-throttle feature.

### CODE REVIEW

**Comment by gordonwang0** [comment]
I didn't add this setting to template.toml, since this should be a rarely-used setting. If needed, we can point to the documentation.

File: cert/aziot-certd-config/src/lib.rs [hunk 1]
```diff
@@ -358,6 +365,7 @@ mod tests {
     fn parse_config() {
         let actual = r#"
 homedir_path = "/var/lib/aziot/certd"
+max_requests = 50
```
Comment (varunpuranik) [inline cert/aziot-certd-config/src/lib.rs:368]: Oh.. so now we need to make the limit 50 so that EdgeAgent's requests succeed?

Comment (gordonwang0) [reply cert/aziot-certd-config/src/lib.rs:368]: This is test code. The default remains 10.

**Review by arsing — APPROVED** [review]

File: cert/aziot-certd-config/src/lib.rs (hunk 1 above)
Comment (varunpuranik) [reply cert/aziot-certd-config/src/lib.rs:368]: ah, yes, missed that.

**Review by varunpuranik — APPROVED** [review]

**Pull request merged**